### PR TITLE
package: fix relative paths and embedded composite actions

### PR DESCRIPTION
**Pull request by ryancragun**
Fix several small issues that make it possible to migrate from the prior
Docker based version to the composite action version

* Check out the action repo into a different directory when building nfpm_template
* Execute nfpm from the root workspace directory to support relative
  paths in configs
* Work around runners bugs when executing composite actions inside the
  scope of other composite actions.

### CODE REVIEW

**Comment by ryancragun** [comment]
Tested this branch in Vault here: https://github.com/hashicorp/vault/actions/runs/9023783609?pr=26905

File: action.yml [hunk 1]
```diff
@@ -184,16 +204,26 @@ runs:
...
+          if ! INPUT_DEPENDS="$2" nfpm_template > "$config_file"; then
+            printf "failed to executing nfpm_template for $1"
+            exit 1
+          fi
+          cat "$config_file"
+          mkdir -p ./out
+          if ! nfpm package -f "$config_file" -p ${1} -t ./out/; then
+            printf "failed to create package with nfpm for $1 using config $(cat $config_file)\n"
+            exit 1
+          fi
```
Comment (shore) [inline action.yml:225]: Nit: I don't think it's helpful to dump the config file a second time, but I do think it's helpful to always dump it.  Maybe remove the conditional and let nfpm's non-zero exit directly halt the script and job?

**Review by shore — APPROVED** [review]
LGTM!

File: action.yml (hunk 1 above)
Comment (ryancragun) [reply action.yml]: I think that was an artifact of me debugging and I missed taking it out!

**Review by shore — APPROVED** [review]
:shipit:

**Pull request merged**